test/specs/components/header.spec.jsx

```diff
@@ -0,0 +1,220 @@
+import sinon from 'sinon';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import TestUtils from 'react-addons-test-utils';
+
+import { scryRenderedDOMComponentsWithId, findRenderedDOMComponentsWithId } from 'test/utils/react';
+
+import { HeaderComponent } from 'components/header.jsx';
+
+const sandbox = sinon.sandbox.create();
+
+const defaultProps = {
+    appState: {
+        widgetOpened: false,
+        settingsEnabled: false,
+        settingsVisible: false
+    },
+    conversation: {
+        messages: []
+    },
+    actions: {
+        showSettings: sandbox.spy(),
+        hideSettings: sandbox.spy(),
+        toggleWidget: sandbox.spy()
+    },
+    ui: {
+        text: {
+            headerText: 'Header',
+            settingsHeaderText: 'Settings'
+        }
+    }
+};
+
+describe('Header', () => {
+    afterEach(() => {
+        sandbox.restore();
+        defaultProps.actions.showSettings = sandbox.spy();
```

Comment:
What if we change `const defaultProps` to `let props` and initialize it in a `beforeEach` like this:

```
    beforeEach( () => {
        props = {
            ...
        };
    });
```

You wouldn't need to reset the spies in this afterEach and it would reduce the possibility of tests interfering with another.